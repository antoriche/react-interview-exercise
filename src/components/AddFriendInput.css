:local(.addFriendInput) {
  border-radius: 0;
  border-color: #ABAAAA;
  border-left: 0;
  border-right: 0;
}

:local(.dropdown) {
  border-radius: 0;
  border-color: #ABAAAA;
  border-left: 0;
  border-right: 0;
}
