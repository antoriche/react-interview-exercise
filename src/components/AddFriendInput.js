import React, { Component, PropTypes } from 'react';
import classnames from 'classnames';
import styles from './AddFriendInput.css';

class AddFriendInput extends Component {

  render () {
    return (
      <div className="input-group">
        <div className="input-group-btn" >
          <button
            type="button"
            onClick={this.switchGender.bind(this)}
            className="btn btn-default dropdown-toggle"
          >
            <i className={classnames('fa', {
              'fa-mars': this.state.gender === 'M',
              'fa-venus': this.state.gender === 'F'
            })} ></i>
          </button>
        </div>
        <input
          type="text"
          autoFocus="true"
          className={classnames('form-control', styles.addFriendInput)}
          placeholder="Type the name of a friend to be added"
          value={this.state.name}
          onChange={this.handleChange.bind(this)}
          onKeyDown={this.handleSubmit.bind(this)}
        />
      </div>
    );
  }

  constructor (props, context) {
    super(props, context);
    this.state = {
      name: this.props.name || '',
      gender: 'M'
    };
  }

  handleChange (e) {
    this.setState({ name: e.target.value });
  }
  switchGender(){
    this.setState({ gender: this.state.gender==='M'?'F':'M' });
  }

  handleSubmit (e) {
    const name = e.target.value.trim();
    if (e.which === 13) {
      this.props.addFriend(name, this.state.gender);
      this.setState({ name: '', gender: 'M' });
    }
  }

}

AddFriendInput.propTypes = {
  addFriend: PropTypes.func.isRequired
};

export default AddFriendInput
