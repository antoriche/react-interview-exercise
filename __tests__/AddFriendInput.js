import React from 'react';
import AddFriendInput from '../src/components/AddFriendInput';
import renderer from 'react-test-renderer';

describe('AddFriendInput', ()=>{
  it('render correctly', () => {
    const component = renderer.create(
      <AddFriendInput
          addFriend = {() => {}}
      />,
    );
    
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('switch gender', () => {
    const component = renderer.create(
      <AddFriendInput
          addFriend = {() => {}}
      />,
    );

    component.getInstance().switchGender();
    
    let tree = component.toJSON();
    // expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree).includes("fa-venus")).toBe(true);
  });

  it('call sumbit function', () => {
    const mockSubmit = jest.fn((name, gender) => {
      expect(name).toBe("Michel");
      expect(gender).toBe('M');
    });
    const component = renderer.create(
      <AddFriendInput
          addFriend = {mockSubmit}
      />,
    );

    component.getInstance().handleSubmit({
      target : {
        value : "Michel"
      },
      which: 13
    });
    
    let tree = component.toJSON();
    expect(mockSubmit.mock.calls.length).toBeGreaterThan(0);
  });
});