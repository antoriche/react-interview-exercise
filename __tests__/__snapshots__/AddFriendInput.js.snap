exports[`AddFriendInput render correctly 1`] = `
<div
  className="input-group">
  <div
    className="input-group-btn">
    <button
      className="btn btn-default dropdown-toggle"
      onClick={[Function]}
      type="button">
      <i
        className="fa fa-mars" />
    </button>
  </div>
  <input
    autoFocus="true"
    className="form-control"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Type the name of a friend to be added"
    type="text"
    value="" />
</div>
`;
